### PR TITLE
fix(qwik-city): cleanup matchRouteRequest

### PR DESCRIPTION
**Pull request by gioboa**
# Overview

Polish matchRouteRequest method and fix typo in a comment.

# Checklist:

- [X] My code follows the [developer guidelines of this project](https://github.com/BuilderIO/qwik/blob/main/CONTRIBUTING.md)
- [X] I have performed a self-review of my own code

### CODE REVIEW

File: packages/qwik-city/buildtime/vite/dev-server.ts [hunk 1]
```diff
@@ -35,23 +35,14 @@ import { matchRoute } from 'packages/qwik-city/runtime/src/route-matcher';
 
 export function ssrDevMiddleware(ctx: BuildContext, server: ViteDevServer) {
   const matchRouteRequest = (pathname: string) => {
+    pathname += ctx.opts.trailingSlash && !pathname.endsWith('/') ? '/' : '';
```
Comment (mhevery) [inline packages/qwik-city/buildtime/vite/dev-server.ts:38]: I am not convinced that this is the same behavior. 
I think the old one checked it twite as it checked the URLs both with and without the trailing slash. The new one does one or the other.

Comment (gioboa) [reply packages/qwik-city/buildtime/vite/dev-server.ts:38]: You are right, if the behavior is to check with or without then this code is not correct. Anyway [trailingSlash](https://qwik.builder.io/docs/advanced/vite/#trailingslash) is a global vite option, do we need to check both cases?

Comment (gioboa) [reply packages/qwik-city/buildtime/vite/dev-server.ts:38]: Btw Array.find() is another improvement

Comment (mhevery) [reply packages/qwik-city/buildtime/vite/dev-server.ts:38]: I would leave the original behavior. The cost of checking the URLs is not that much.

Comment (gioboa) [reply packages/qwik-city/buildtime/vite/dev-server.ts]: With this code I removed one for loop and the behaviour remains the same

**Review by mhevery — APPROVED** [review]

**Pull request merged**